lib/api/BackbeatRequest.js

```diff
@@ -33,18 +33,25 @@ class BackbeatRequest {
     }
 
     /**
-     * Parse the route details for any of the retry routes.
+     * Parse the route details for any of the crr routes (retry, pause, resume).
      * @param {Array} parts - The route schema split by '/'
      * @param {String} query - The query string.
      * @return {undefined}
      */
-    _parseRetryRoutes(parts, query) {
-        this._routeDetails.extension = parts[0];
-        this._routeDetails.status = parts[1];
-        this._routeDetails.bucket = parts[2];
-        this._routeDetails.key = parts[3];
-        this._routeDetails.versionId = parts[4];
-        this._routeDetails.marker = querystring.parse(query).marker;
+    _parseCRRRoutes(parts, query) {
+        if (parts[1] && parts[1] === 'failed') {
+            this._routeDetails.extension = parts[0];
```

Comment:
As a future refactor, we should only parse path components up to what allows to dispatch to the proper route, and defer the responsability of parsing the rest of the path components to the specific route implementation, as it's going to become more spaghetti-style each time we add a new route.